Sources/PrimerSDK/Classes/User Interface/OAuth/OAuthViewController.swift

```diff
@@ -238,46 +271,49 @@ class WebViewController: UIViewController, WKNavigationDelegate {
         ]
 
         if let url = navigationAction.request.url, let host = url.host, allowedHosts.contains(host) {
-
             let val = url.queryParameterValue(for: "token")
+            
+            if (val ?? "").isEmpty || val == "undefined" {
+                let err = PrimerError.clientTokenNull
+                klarnaWebViewCompletion?(nil, err)
+                decisionHandler(.cancel)
+                return
+            }
 
             log(logLevel: .info, message: "🚀🚀 \(url)")
             log(logLevel: .info, message: "🚀🚀 token \(val)")
             
             let state: AppStateProtocol = DependencyContainer.resolve()
 
+            // FIXME: WebView should be agnostic of state and not set it. Debug what happens on
+            // PayPal case befire removing.
             state.authorizationToken = val
+            klarnaWebViewCompletion?(val, nil)
 
             log(logLevel: .info, message: "🚀🚀🚀 \(state.authorizationToken ?? "n/a")")
 
+            // Cancels navigation
             decisionHandler(.cancel)
-
-            let settings: PrimerSettingsProtocol = DependencyContainer.resolve()
-            let routerDelegate: RouterDelegate = DependencyContainer.resolve()
-            let router = routerDelegate as! Router
-            let rootViewController = router.root
-            
-            if settings.hasDisabledSuccessScreen == false && settings.isInitialLoadingHidden == true {
-                let theme: PrimerThemeProtocol = DependencyContainer.resolve()
-                rootViewController?.mainView.backgroundColor = theme.colorTheme.main1
-                if #available(iOS 11.0, *) {
-                    (rootViewController?.children.first as? OAuthViewController)?.indicator.isHidden = false
-                }
-                
-            } else if settings.hasDisabledSuccessScreen && settings.isInitialLoadingHidden {
-                UIView.animate(withDuration: 0.3) {
-                    (rootViewController?.presentationController as? PresentationController)?.blurEffectView.alpha = 0.0
-                }
-            }
             
-            dismiss(animated: true, completion: nil)
-
-            return
+        } else {
+            // Allow navigation to continue
+            decisionHandler(.allow)
         }
-
-        decisionHandler(.allow)
     }
-
+    
+    func webView(_ webView: WKWebView, didFailProvisionalNavigation navigation: WKNavigation!, withError error: Error) {
+        let nsError = error as NSError
+        if !(nsError.code == -1002) {
```

Comment:
This will throw whatever navigational error Apple throws in a webview. Previously these errors were completely ignored, as we weren't conforming on this delegate method. Code -1002 is for a `Bad URL error` and `bankid://` is considered a bad URL format. Therefore, this error needs to be ignored. The rest of the errors (e.g. a timeout error) shouldn't be ignored. So what happens now is that these errors will return an error through the delegate method and dismiss the webview. 

I have to take a look on the rest of the Apple defined errors cause there may be some others that need to be excluded as well.